### PR TITLE
Support for tf df trees #181

### LINKED ISSUE #181: Support for tensorflow - decision forest
https://blog.tensorflow.org/2021/05/introducing-tensorflow-decision-forests.html

### PR DESCRIPTION
**Pull request by tlapusan**
@parrt it contains the visualisations for Random Forest algo. The tensorflow DF contains more algorithms for decision forests and we can add them also in the future.

close #181

### CODE REVIEW

File: testing/testlib/models/test_decision_tree_spark_classifier.py [hunk 1]
```diff
@@ -21,10 +21,10 @@ def tree_model() -> (DecisionTreeClassificationModel):
 
 
 @pytest.fixture()
-def spark_dtree(tree_model, dataset_spark) -> ShadowSparkTree:
+def spark_dtree(tree_model, dataset_spark_tf) -> ShadowSparkTree:
```
Comment (parrt) [inline testing/testlib/models/test_decision_tree_spark_classifier.py:25]: Should this really be `tf`? Seems like it should be just spark or is somehow spark using tf underneath?

**Comment by parrt** [comment]
hooray!!!  The `tfdf` prefix looks very much like the common term `tfidf`... Perhaps we change that to simply prefix of `tf` everywhere?

**Comment by parrt** [comment]
There are a couple of changes such as in interpretation.py that might affect other targets. Have you run through and checked the other kinds of trees and run tests?

File: testing/testlib/models/test_decision_tree_spark_classifier.py (hunk 1 above)
Comment (tlapusan) [reply testing/testlib/models/test_decision_tree_spark_classifier.py:25]: dataset_spark_tf it's a fixture for the dataset used in training for Spark and TF. I used the same dataset for testing for both spark and TF. Looking in the conftest.py I see we have different dataset fixtures for each library...don't remember exactly why.. maybe we needed a different format.

**Comment by tlapusan** [comment]
> There are a couple of changes such as in interpretation.py that might affect other targets. Have you run through and checked the other kinds of trees and run tests?

yes, I spent some time for these changes and ran/tested all of them :)

**Comment by tlapusan** [comment]
> hooray!!! The `tfdf` prefix looks very much like the common term `tfidf`... Perhaps we change that to simply prefix of `tf` everywhere?

agree, I will rename them soon.

**Comment by parrt** [comment]
Thanks, @tlapusan !

**Pull request merged**